server/utils.go

```diff
@@ -77,6 +80,38 @@ func SendHTTPRequest(ctx context.Context, client http.Client, method, url string
 	return resp.StatusCode, nil
 }
 
+// SendHTTPRequestWithRetries - prepare and send HTTP request, retrying the request if within the client timeout
+func SendHTTPRequestWithRetries(ctx context.Context, client http.Client, method, url string, userAgent UserAgent, payload any, dst any, handleError ErrorHandlerFunc) (code int, err error) {
+	// default no retry if no timeout set
+	if client.Timeout == 0 {
+		return SendHTTPRequest(ctx, client, method, url, userAgent, payload, dst)
+	}
+
+	// Create a context with a timeout as configured in the http client
+	ctxTimeout, cancel := context.WithTimeout(context.Background(), client.Timeout)
+	defer cancel()
+
+	attempts := 0
+	retryWaitTime := 1 * time.Second
+	for {
+		select {
+		case <-ctxTimeout.Done():
```

Comment:
fyi, this will not interrupt a running request because it's blocking in the `default:` part, the current code works just like a for loop without select.